packages/contracts-bedrock/src/Safe/DeputyGuardianModule.sol

```diff
@@ -0,0 +1,112 @@
+import { OptimismPortal2 } from "src/L1/OptimismPortal2.sol";
+import { IDisputeGame } from "src/dispute/interfaces/IDisputeGame.sol";
+import { ISemver } from "src/universal/ISemver.sol";
+
+import "src/libraries/DisputeTypes.sol";
+
+/// @title DeputyGuardianModule
+/// @notice This module is intended to be enabled on the Security Council Safe, which will own the Guardian role in the
+///         SuperchainConfig contract. The DeputyGuardianModule should allow a Deputy Guardian to administer any of the
+///         actions that the Guardian is authorized to take. The security council can revoke the Deputy Guardian's
+///         authorization at any time by disabling this module.
+contract DeputyGuardianModule is ISemver {
+    /// @notice The Safe contract instance
+    Safe internal immutable SAFE;
+
+    /// @notice The SuperchainConfig's address
+    SuperchainConfig internal immutable SUPERCHAIN_CONFIG;
+
+    /// @notice The deputy guardian's address
+    address internal immutable DEPUTY_GUARDIAN;
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    // Constructor to initialize the Safe and baseModule instances
+    constructor(Safe _safe, SuperchainConfig _superchainConfig, address _deputyGuardian) {
+        SAFE = _safe;
+        SUPERCHAIN_CONFIG = _superchainConfig;
+        DEPUTY_GUARDIAN = _deputyGuardian;
+    }
+
+    /// @notice Getter function for the Safe contract instance
+    /// @return safe_ The Safe contract instance
+    function safe() public view returns (Safe safe_) {
+        safe_ = SAFE;
+    }
+
+    /// @notice Getter function for the SuperchainConfig's address
+    /// @return superchainConfig_ The SuperchainConfig's address
+    function superchainConfig() public view returns (SuperchainConfig superchainConfig_) {
+        superchainConfig_ = SUPERCHAIN_CONFIG;
+    }
+
+    /// @notice Getter function for the deputy guardian's address
+    /// @return deputyGuardian_ The deputy guardian's address
+    function deputyGuardian() public view returns (address deputyGuardian_) {
+        deputyGuardian_ = DEPUTY_GUARDIAN;
+    }
+
+    function _onlyDeputyGuardian() internal view {
+        require(msg.sender == DEPUTY_GUARDIAN, "DeputyGuardianModule: Only the deputy guardian can call this function.");
+    }
+
+    /// @notice Calls to the Security Council's `execTransactionFromModule()`, with the arguments
+    ///      necessary to call `pause()` on the `SuperchainConfig` contract.
+    ///      Only the deputy guardian can call this function.
+    function pause() external {
+        _onlyDeputyGuardian();
+        bytes memory data = abi.encodeWithSelector(SUPERCHAIN_CONFIG.pause.selector, "");
+
+        (bool success, bytes memory returnData) =
+            SAFE.execTransactionFromModuleReturnData(address(SUPERCHAIN_CONFIG), 0, data, Enum.Operation.Call);
+        require(success, string(returnData));
```

Comment:
Do we want this change to follow the standard format?
```suggestion
        require(success, string.concat("DeputyGuardianModule: ", string(returnData)));
```

Reply:
Is that standard?

In most cases when we make a high-level solidity-syntax call to an external contract, it'll just re-throw/bubble up the error message without adding a prefix.

Looking around for standard, we do an entirely different thing in the liveness module: 

https://github.com/ethereum-optimism/optimism/blob/maur/guardian-module/packages/contracts-bedrock/src/Safe/LivenessModule.sol#L173-L180

But in hindsight, I dislike that approach because it's not providing any real insight into why the call made by the Safe failed, ie. it would bury the [GS104](https://github.com/safe-global/safe-contracts/blob/v1.4.0/contracts/base/ModuleManager.sol#L88) message if that was thrown.

Reply:
Ah, I just assumed it was standard based on the [style guide](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/STYLE_GUIDE.md#errors):

> Error strings MUST have the format `"{ContractName}: {message}"` where `message` is a lower case string.

I agree the liveness module approach isn't ideal for the reason you mentioned and think we should change that.

We did [recently discuss](https://oplabs-pbc.slack.com/archives/C03NW5QRXDF/p1709655647616989?thread_ts=1709643492.100419&cid=C03NW5QRXDF)  removing new `require` statements altogether in favor of custom errors, which also an alternative here (and would need a style guide update too)

Reply:
I see. My somewhat pedantic point was that `"{ContractName}: {message}"` is standard, but your proposal would not be standard because it would revert with:
`"DeputyGuardianModule: {ContractName}: {message}"`

But just using custom errors seems like a good way to circumvent this discussion.

Reply:
OK, I added a custom error, so we're now reverting with `revert TransactionExecutionFailed(string(returnData))`.